packages/api/src/makeMergedSchema/rootSchema.ts

```diff
@@ -1,6 +1,6 @@
 /* eslint-disable @typescript-eslint/ban-ts-comment */
 import type { GlobalContext } from 'src/globalContext'
-import { PrismaClient } from '@prisma/client'
+const { prismaVersion } = require('@prisma/client')
```

Comment:
Is there a reason why this line is not:

`import { prismaVersion } =  from '@prisma/client'`

?

Reply:
I think @AntonioMeireles already tried that over here: https://github.com/AntonioMeireles/redwood/commit/3283d28b8d03223e6f6dfaacc8d4fa76d271bc2b

Reply:
Not sure why it didn't work though...

Reply:
Yeah that doesn't work somehow. That was the reason I originally imported the `prismaClient` and fetched the version from there.

Reply:
yes... for reasons that i don't get it simply does not work as the typescript compiler doesn't like it... 

build busts with ... 
```
6:27:15 PM] Building project '/Users/am/hacks/homeWare/redwood/packages/api/tsconfig.json'...

packages/api/src/makeMergedSchema/rootSchema.ts:4:1 - error TS6133: 'PrismaClient' is declared but its value is never read.

4 import { PrismaClient } from '@prisma/client'
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

packages/api/src/makeMergedSchema/rootSchema.ts:57:28 - error TS2304: Cannot find name 'prismaVersion'.

57       prismaVersion: () => prismaVersion.client,
                              ~~~~~~~~~~~~~
```

with `const { prismaVersion } = require('@prisma/client')` it just builds.

Reply:
frankly my ts foo is not deep enough to spot why this has to be this way .

Reply:
Does this

https://github.com/prisma/prisma-client-js/issues/573

help?

```
/**
 * Query Engine version: 4ff8379527ec7797e7bb5b55d374f82f5812a6f9
 * Prisma Client JS version: 2.0.0-alpha.913
 */
exports.version = {
  engine: "4ff8379527ec7797e7bb5b55d374f82f5812a6f9",
  client: "2.0.0-alpha.913"
}
```

Maybe

```
import { version} from '@prisma/client'

...

prismaVersion: () => version.client,

```

Reply:
no. they renamed a bit later - https://github.com/prisma/prisma-client-js/commit/ce3e7effb2d0d3a865d0beafec78f73abddf89cf  (to `prismaVersion`)
It's  worth nothing that expected code to work *only* fails over typescript. as javascript it builds/runs nicely. when in typescript somehow the type machinery gets fooled.

Reply:
(fixed above the commit where they actually `s/version/prismaVersion/`)

Reply:
> no. they renamed a bit later

🙄  oh did they? 

sometimes i refuse to give up. i tried!

Reply:
I think the reason why it probably didn't work could be because the client isn't generated. I think we should inline the entire `require` statement.

Reply:
hang on, a bit. 

error snippet above was wrong. 
real error when trying to do things the expected way is ... 
```
6:47:40 PM] Building project '/Users/am/hacks/homeWare/redwood/packages/api/tsconfig.json'...

packages/api/src/makeMergedSchema/rootSchema.ts:4:10 - error TS2305: Module '"../../../../node_modules/@prisma/client"' has no exported member 'prismaVersion'.

4 import { prismaVersion } from '@prisma/client'
           ~~~~~~~~~~~~~

[6:47:44 PM] Project 'packages/api-server/tsconfig.json' is up to date because newest input 'packages/api-server/src/error.ts' is older than oldest output 'packages/api-server/dist/error.d.ts.map'


Found 1 error.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 
```

googling a bit i get https://github.com/prisma/prisma/issues/2584 which may or not be related ...
it references symlinks which rw workspaces abuse a lot . again ts / yarn not my core domain at all :-)

Reply:
As I was digging through

Reply:
> I think the reason why it probably didn't work could be because the client isn't generated.

re-reading prisma/prisma#2584  it may be just that, yes!

Reply:
Yeah, looking at prisma client's index.d.ts for the rw framework project it's basically just 
```
export declare const PrismaClient: any
export declare type PrismaClient = any
```

But if you look at it in a RW project where the client has been generated index.d.ts has a lot more, like
```
/**
 * Prisma Client JS version: 2.9.0
 * Query Engine version: 369b3694b7edb869fad14827a33ad3f3f49bbc20
 */
export declare type PrismaVersion = {
  client: string
}

export declare const prismaVersion: PrismaVersion 
```